### DISCUSSION TITLE
PROBLEM: availableAnimation doesnt return any animation.

### DISCUSSION THREAD

**Discussion by phovorka** [google/model-viewer#4608] [General]
Hi,

i have a problem with detecting animation . Even if i have GLB file with animation. "availableAnimation" returns length 0. Is there any issue?

> let animations = modelViewer.availableAnimations;
> let numberOfAnimations = animations.length;

> console.log(animations);
> if (numberOfAnimations) {
>   animToolbar.style.display = "block";
> } else {
>   animToolbar.style.display = "none";
> }

**Comment by elalish** [comment]
Did you forget to wait for the async "load" event?

**Reply by phovorka** [reply]
async is not solving this issue (i tried with  DOMContentLoaded) . INTERESTING:  once i load model with HTML attribut src it works fine, but when i load model with JS (modelViewer.src) it doesnt work ..... hmmm but i need to load it via JS

**Reply by elalish** [reply]
No, you specifically need *our* load event: https://modelviewer.dev/docs/index.html#entrydocs-loading-events-load

**Reply by phovorka** [reply]
aaaah. This seems promising. Thx Emmett!
